client/src/app/dashboard/table-all/table-all.component.ts

```diff
@@ -16,53 +16,152 @@
-    return numSelected === numRows;
-  }
-
-  masterToggle() {
-    this.isAllSelected() ?
-        this.selection.clear() :
-        this.data_array.forEach((row) => {this.selection.select(row); });
-  }
+
+    @ViewChild(MatPaginator) paginator: MatPaginator;
+
+    ngOnInit() {
+        this.dataSource.paginator = this.paginator;
+    }
+
+    // Function to Delete the data from datasource
+    openDeleteDialog(i: number, instance: Instance, id: string) {
+        const dialogRef = this.dialog.open(DeleteDialogComponent, {
+            width: '250px',
+            data: { name: instance.name }
+        });
+
+        dialogRef.afterClosed().subscribe(result => {
+            console.log('delete state', instance.instanceState);
+            if (result === 'Confirm' && instance.instanceState === 'Running') {
+                console.log('alert working');
+                alert('Please Stop the Instance before you try to delete');
+                console.log('data', this.dataSource.data);
+
+            } else {
+                this.apiService.deleteInstance(id).subscribe((deleteResult: number) => {
+                    console.log('result', deleteResult);
+                }, err => {
+                    console.log('error delete Instance');
+                });
+                this.removeAt(i);
+
+            }
+            this.dialogResult = result;
+        });
+    }
+
+    removeAt(index: number) {
+        this.dataSource.data.splice(index, 1);
+        this.table.renderRows();
+    }
+
+    applyFilter(filterValue: string) {
+        this.dataSource.filter = filterValue.trim().toLowerCase();
+    }
+
+    // Function to add the data into dataSource
+    openAddDialog() {
+        const instance: Instance = {};
+        instance.componentType = this.type;
+        const dialogRef = this.dialog.open(AddDialogComponent, {
+            width: '300px',
+
+        });
+
+        dialogRef.afterClosed().subscribe(dialogResult => {
+            this.apiService.postInstance(instance.componentType, dialogResult.name).subscribe((result: Instance) => {
+                this.dataSource.data.push({
+                    id: result.id,
+                    host: result.host,
+                    portNumber: result.portNumber,
+                    name: dialogResult.name,
+                    instanceState: result.instanceState,
+                    componentType: instance.componentType
+                });
+                this.table.renderRows();
+            }, err => {
+
+                console.log('error receiving data for crawler');
+            });
+        });
+    }
+
+    // Function to 'start' the instance
+    public startInstance(id: string): void {
+
+        this.apiService.startInstance(id).subscribe((result: number) => {
+            console.log('result', result);
+        }, err => {
+            console.log('error start Instance', err);
+        });
+    }
+
+    // Function to 'stop' the instance
```

Comment:
you are describing a function with it's name, please try to find a bit more speaking descriptions than this.
also please use javadoc comment style /** */ instead of // to document methods.

Reply:
ok